mne/io/meas_info.py

```diff
@@ -2906,6 +2906,10 @@ def _ensure_infos_match(info1, info2, name, *, on_mismatch='raise'):
         raise ValueError(f'{name}.info[\'sfreq\'] must match')
     if set(info1['ch_names']) != set(info2['ch_names']):
         raise ValueError(f'{name}.info[\'ch_names\'] must match')
```

Comment:
I kept this to enable a more precise error message. This way it should be faster to understand for the user whether some channels are missing or whether the channel order is wrong.